src/cache-restore.ts

```diff
@@ -36,12 +39,13 @@ export const restoreCache = async (
     );
   }
 
-  const primaryKey = `node-cache-${platform}-${packageManager}-${fileHash}`;
+  const keyPrefix = `node-cache-${platform}-${packageManager}-${stepId}-${nodeVersion}`;
```

Comment:
Curious - Why stepId ?. In my understanding / experience the cache is totally shareable between steps.